cli/tools/bundle/mod.rs

```diff
@@ -0,0 +1,868 @@
+  ) -> Result<(), AnyError> {
+    let mut graph_permit =
+      self.module_graph_container.acquire_update_permit().await;
+    let graph: &mut deno_graph::ModuleGraph = graph_permit.graph_mut();
+    self
+      .module_load_preparer
+      .prepare_module_load(
+        graph,
+        specifiers,
+        PrepareModuleLoadOptions {
+          is_dynamic: false,
+          lib: TsTypeLib::default(),
+          permissions: self.permissions.clone(),
+          ext_overwrite: None,
+          allow_unknown_media_types: false,
+          skip_graph_roots_validation: true,
+        },
+      )
+      .await
+      .inspect_err(|_e| {
+        // eprintln!(
+        //   "{}: error preparing module load: {:?}",
+        //   deno_terminal::colors::red("ERROR"),
+        //   e
+        // );
+      })?;
+    // eprintln!("prepared module load");
+    graph_permit.commit();
+    Ok(())
+  }
+
+  async fn bundle_load(
+    &self,
+    specifier: &str,
+    resolve_dir: &str,
+  ) -> Result<Option<(Vec<u8>, esbuild_rs::BuiltinLoader)>, AnyError> {
+    log::debug!(
+      "{}: {:?} {:?}",
+      deno_terminal::colors::magenta("bundle_load"),
+      specifier,
+      resolve_dir
+    );
+
+    let resolve_dir = Path::new(&resolve_dir);
+    let specifier = deno_core::resolve_url_or_path(specifier, resolve_dir)?;
+
+    // let (code, loader) =
+    //   if let Some((code, loader)) = self.load_from_graph(&specifier)? {
+    //     (code, loader)
+    //   } else {
+    //     log::debug!(
+    //       "{}: {}",
+    //       deno_terminal::colors::cyan("module is none"),
+    //       specifier
+    //     );
+    //     self.prepare_module_load(&[specifier.clone()]).await?;
+    //     let Some((code, loader)) = self.load_from_graph(&specifier)? else {
+    //       return Ok(None);
+    //     };
+    //     (code, loader)
+    //   };
+    // Ok(Some((code, loader)))
+
+    let (specifier, loader) = if let Some((specifier, loader)) =
+      self.specifier_and_type_from_graph(&specifier)?
+    {
+      (specifier, loader)
+    } else {
+      log::debug!(
+        "{}: no specifier and type from graph for {}",
+        deno_terminal::colors::yellow("warn"),
+        specifier
+      );
+
+      let (media_type, _) =
+        deno_media_type::resolve_media_type_and_charset_from_content_type(
+          &specifier, None,
+        );
+      if media_type == deno_media_type::MediaType::Unknown {
+        return Ok(None);
+      }
+      (specifier, media_type_to_loader(media_type))
+    };
+    let loaded = self.module_loader.load(
+      &specifier,
+      None,
+      false,
+      deno_core::RequestedModuleType::None,
+    );
+
+    match loaded {
+      deno_core::ModuleLoadResponse::Sync(module_source) => {
+        Ok(Some((module_source?.code.as_bytes().to_vec(), loader)))
+      }
+      deno_core::ModuleLoadResponse::Async(pin) => {
+        let pin = pin.await?;
+        Ok(Some((pin.code.as_bytes().to_vec(), loader)))
+      }
+    }
```

Comment:
Hmmm... might be nice if we figured out how to remove `deno_core` from here.

Reply:
IMO this is fine since we use `deno_core::ModuleLoader` here